dev-tools/mage/gotool/go.go

```diff
@@ -93,14 +93,19 @@ func ListTestFiles(pkg string) ([]string, error) {
 	return getLines(callGo(nil, "list", "-f", tmpl, pkg))
 }
 
-// ListModulePath returns the path to the module in the cache.
-func ListModulePath(pkg string) (string, error) {
-	const tmpl = `{{.Dir}}`
+// ListModulePath returns the path to the specified module.
+//
+// Additional parameters like "-mod=vendor" may be passed
+// to control behaviour.
+func ListModulePath(pkg string, args ...string) (string, error) {
 	env := map[string]string{
-		// make sure to look in the module cache
+		// make sure to look in the module cache,
+		// unless otherwise configured in args.
 		"GOFLAGS": "",
```

Comment:
We can set `GOFLAGS` to `-mod=vendor`, so there is no need to append more args to the existing command.

I would introduce a new function `ListModuleFromVendor` so there is no need for passing extra params.

Reply:
I've renamed `ListModulePath` to `ListModuleCacheDir`, and introduced `ListModuleVendorDir`. In dev-tools/mage, I introduced an unexported function `listModuleDir` which calls one or the other depending on the value of `UseVendor`. In some places we still call `ListModuleCacheDir` explicitly, e.g. when copying stuff into the vendor directory.